feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java

```diff
@@ -20,165 +34,196 @@
-
-        JSONObject parameters = new JSONObject();
-        parameters.put("name", name);
-        parameters.put("contact_name", uuid);
-        parameters.put("currency_id", currencyId);
-        String jsonInputString = parameters.toString();
-
-        try (OutputStream os = con.getOutputStream()) {
-            byte[] inputString = jsonInputString.getBytes(UTF_8);
-            os.write(inputString, 0, inputString.length);
-        }
-        con.getInputStream();
-    }
-
-    public void update_customer(OpenMRSPatientFullRepresentation patientFR, String auth) throws Exception {
-        String name = patientFR.getPerson().getPreferredName().getPreferredFullName();
-        String uuid = getuuid(patientFR);
-
-        String id = this.get_list_customers(uuid, auth);
-        HttpURLConnection con = getConnection("customers/" + id, "PUT", auth);
-
-        JSONObject parameters = new JSONObject();
-        parameters.put("name", name);
-        parameters.put("currency_id", currencyId);
-        String jsonInputString = parameters.toString();
-
-        try (OutputStream os = con.getOutputStream()) {
-            byte[] inputString = jsonInputString.getBytes(UTF_8);
-            os.write(inputString, 0, inputString.length);
-        }
-        con.getInputStream();
-    }
-
-    public String get_list_customers(String uuid, String auth) throws URISyntaxException, JSONException, IOException {
-        CloseableHttpClient httpClient = HttpClients.createDefault();
+		this.auth = craterLogin.getToken();
+	}
+
+	@Value("${con.content_type}")
+	String content_type;
+
+	@Value("${con.accept}")
+	String accept;
+
+	@Value("${con.setDoOutput}")
+	boolean do_Output;
+
+	@Value("${con.setDoInput}")
+	boolean do_Input;
+
+	@Value("${crater.currency.id}")
+	String currencyId;
+
+	@Value("${crater.url}")
+	String crater_url;
+
+	@Value("${crater.company}")
+	String company;
+
+
+	private static final Logger logger = LoggerFactory.getLogger(CraterAPIClient.class);
+
+	public boolean login_verification(String auth) throws Exception {
+		CloseableHttpClient httpClient = HttpClients.createDefault();
+		HttpGet request = new HttpGet(crater_url + "/api/v1/auth/check");
+
+		request.addHeader("Content-Type", content_type);
+		request.addHeader("Accept", accept);
+		request.addHeader("company", company);
+		request.addHeader("Authorization", "Bearer " + auth);
+
+		try (CloseableHttpResponse response = httpClient.execute(request)) {
+			HttpEntity entity = response.getEntity();
+			String result = EntityUtils.toString(entity);
+			httpClient.close();
+			response.close();
+			return ((result.charAt(0) == '1') ? true : false);
+		}
+	}
+
+	public void create_customer(OpenMRSPatientFullRepresentation patientFR, String auth) throws Exception {
+		String name = patientFR.getPerson().getPreferredName().getPreferredFullName();
+		String uuid = getuuid(patientFR);
+		List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>();
```

Comment:
The code for constructing nameValuePairs in create_cust() and update_cust() is the same. Plz extract it into a constructCustomerFromOpenmrsPatient() method and use that across. That way its also easier to write a unit test for that function.